### PR TITLE
prov/efa: handle removed peer

### PR DESCRIPTION
**Pull request by wzamazon**
This PR contains a series of patches that handles the case of a peer was removed from address vector.

### CODE REVIEW

File: prov/efa/src/rxr/rxr_pkt_cmd.c [hunk 1]
```diff
@@ -671,9 +671,155 @@ void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
...
+ *
+ * If the error is RNR, this function will queue a packet if necessary. It
+ * determines whether to queue a RNR packet using the following rule:
+ *
+ *	In general, RNR packet will be queued and resend by the progress enginee.
+ *
+ *	It is NOT queued only if application want to receive an RNR error CQ entry
+ *	for an send operation.
+ *
+ * For other types of error, the function will write cq/eq error entry then
```
Comment (bwbarrett) [inline prov/efa/src/rxr/rxr_pkt_cmd.c:683]: you forgot to mention that there are packets that will never return to application here.

**Review by bwbarrett — APPROVED** [review]
On the second patch, you have a typo in the second paragraph: "This patch implemented that me by introducing..."

File: prov/efa/src/rxr/rxr_pkt_cmd.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_cmd.c:683]: Sorry. I am not sure I understand. You mean for some packet, the device will not return completion to EFA provider? or EFA provider will not return completion to application?

Comment (bwbarrett) [reply prov/efa/src/rxr/rxr_pkt_cmd.c:683]: I liked your description in the commit message better than this comment in terms of figuring out when the events occurred.

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_cmd.c]: Thanks for clarifying! I update the comments by rewording the commit message.

**Pull request merged**